daml-assistant/exe/DAML/Assistant.hs

```diff
@@ -66,74 +93,117 @@ main = displayErrors $ do
+            isLatest
+                | Just (DamlAssistantSdkVersion v) <- envDamlAssistantSdkVersion =
+                    sdkVersion > v
+                | otherwise =
+                    True
+            options = InstallOptions
+                { iTargetM = Nothing
+                , iQuiet = QuietInstall False
+                , iActivate = ActivateInstall isLatest
+                , iForce = ForceInstall False
+                , iSetPath = SetPath True
+                }
+            installEnv = InstallEnv
+                { options = options
+                , damlPath = envDamlPath
+                , targetVersionM = Just sdkVersion
+                , projectPathM = Nothing
+                , output = hPutStrLn stderr
+                    -- Print install messages to stderr since the install
+                    -- is only happening because of some other command,
+                    -- and we don't want to mess up the other command's
+                    -- output / have the install messages be gobbled
+                    -- up by a pipe.
+                }
+        versionInstall installEnv sdkVersion
+        pure env { envSdkPath = Just (defaultSdkPath envDamlPath sdkVersion) }
+
+    else
+        pure env
+
+handleCommand :: Env -> Command -> IO ()
+handleCommand env@Env{..} = \case
+
+    Builtin Version -> do
+        installedVersionsE <- tryAssistant $ getInstalledSdkVersions envDamlPath
+        defaultVersionM <- tryAssistantM $ getDefaultSdkVersion envDamlPath
+
+        let asstVersion = unwrapDamlAssistantSdkVersion <$> envDamlAssistantSdkVersion
+            envVersions = catMaybes
+                [ envSdkVersion
+                , envLatestStableSdkVersion
+                , asstVersion
+                ]
 
-        Builtin Version -> do
-            installedVersionsE <- tryAssistant $ getInstalledSdkVersions envDamlPath
-            defaultVersionM <- tryAssistantM $ getDefaultSdkVersion envDamlPath
+            isInstalled =
+                case installedVersionsE of
+                    Left _ -> const True
+                    Right vs -> (`elem` vs)
+
+            versionAttrs v = catMaybes
+                [ "active"
+                    <$ guard (Just v == envSdkVersion)
+                , "default"
+                    <$ guard (Just v == defaultVersionM)
+                , "assistant"
+                    <$ guard (Just v == asstVersion)
+                , "latest release"
+                    <$ guard (Just v == envLatestStableSdkVersion)
+                , "not installed"
+                    <$ guard (not (isInstalled v))
+                ]
 
-            let asstVersion = unwrapDamlAssistantSdkVersion <$> envDamlAssistantSdkVersion
-                envVersions = catMaybes
-                    [ envSdkVersion
-                    , envLatestStableSdkVersion
-                    , asstVersion
+            -- | Workaround for Data.SemVer old unfixed bug (see https://github.com/brendanhay/semver/pull/6)
```

Comment:
Do we still need this now that there is a new semver release?

Reply:
I'll test & remove this in a separate PR :)